aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -425,19 +426,22 @@ public void observe(
     @Override
     public void clear(@NonNull Action onComplete,
                       @NonNull Consumer<DataStoreException> onError) {
-        beforeOperation(() -> {
-            orchestrator.stop();
-            sqliteStorageAdapter.clear(onComplete, onError);
-        });
+        // We shouldn't call call beforeOperation when clearing the DataStore. The
+        // only thing we have to wait for is the category initialization latch.
+        try {
+            categoryInitializationsPending.await();
+        } catch (InterruptedException exception) {
+            LOG.warn("Execution interrupted while waiting for DataStore to be initialized.");
+        }
+        orchestrator.stop();
+        sqliteStorageAdapter.clear(() -> {
+            initializeOrchestrator().subscribe();
+            onComplete.call();
```

Comment:
Should `onComplete.call()` be fired _as a result_ of the subscription, when the initialization of the orchestrator completes, later? That is,
```java
initializeOrchestrator().subscribe(onComplete::call);
```

Reply:
Done